shared/js/intl_helper.js

```diff
@@ -0,0 +1,148 @@
+(function(global) {
+  'use strict';
+
+  /*************************
+
+  IntlHelper.defineObject('shortDate', 'datetime', {
+    year: 'numeric',
+    month: 'long',
+  });
+
+  function setValue() {
+    var formatter = IntlHelper.getObject('shortDate');
+    console.log(formatter.format(new Date()));
+  }
+
+  setValue();
+
+  IntlHelper.on('shortDate', setValue);
+  
+   ****************************/
+  var objectCache;
+
+  var knownObjects = [
+    'datetime',
+    'number',
+  ];
+
+  global.IntlHelper = {
+    defineObject: function(name, type, options) {
+      if (knownObjects.indexOf(type) === -1) {
+        throw new Error('Unknown type: ' + type);
+      }
+
+      if (!objectCache) {
+        objectCache = new Map();
+      }
+
+      if (objectCache.has(name)) {
+        throw new Error(
+          'Intl Object with name "' + name + '" already exists');
+      }
+
+      objectCache.set(name, {
+        options,
+        type,
+        object: undefined,
+        listeners: new Set()
+      });
+    },
+
+    getObject: function(name) {
+      if (!objectCache.has(name)) {
+        throw new Error('Intl Object with name "' + name + '" is not defined');
+      }
+      const obj = objectCache.get(name);
+
+      if (obj.object === undefined) {
+        switch (obj.type) {
+          case 'number':
+            obj.object = Intl.NumberFormat(
+              navigator.languages,
+              obj.options
+            );
+            break;
+          case 'datetime':
+            const options = Object.assign({}, obj.options);
+            if (options.hour) {
+              options.hour12 = navigator.mozHour12;
+            }
+            console.log('new Format');
+            console.log(options);
+            obj.object = Intl.DateTimeFormat(
+              navigator.languages,
+              options
+            );
+            break;
+          default:
+            throw new Error('Unknown type: ' + obj.type);
+        }
+      }
+      return obj.object;
+    },
+    on: function(names, cb) {
+      if (typeof names === 'string') {
+        names = [names];
```

Comment:
Oh, I think I see it now:  `emit` takes the `type`, but `on` takes the `name`.  So instead of:

    IntlHelper.on('datetimechange', ...);

you now do:

    IntlHelper.on(['shortDate', 'longDate'], ...);

I'm not sure I have an opinion on this, but I do have two comments:

- you should make sure all `names` have been defined, otherwise I think line 95 will break (`obj.listeners` wil be undefined),

- you might want to pass the name of the object to the callback.  Looking at the sample code in the comment at the top of the file, what happens when I listen to more than one object changes?

```javascript
IntlHelper.on(['shortDate', 'longDate'], setValue);

function setValue() {
  var formatter = IntlHelper.getObject('shortDate');      // ← shortDate is hardcoded here.
  console.log(formatter.format(new Date()));
}
```